session/cookie.js

```diff
@@ -8,145 +8,181 @@
+      this.path = options.path || '/'
+      this.sameSite = options.sameSite
+
+      this.signed = options.signed // FIXME: how this is used??
+      this.encode = options.encode // FIXME: is this used / real ??
+    } else {
+      this.path = '/'
+      this.httpOnly = true
     }
   }
 
-  if (this.originalMaxAge === undefined || this.originalMaxAge === null) {
-    this.originalMaxAge = this.maxAge
+  /**
+   * Initialize a new `Cookie` using stored cookie data.
+   * @param {CookieOptions & {expires?: string, originalMaxAge?: number}} data
+   * @returns {Cookie}
+   * @protected
+   */
+  static fromJSON(data) {
+    console.log(`Cookie.fromJSON: ${JSON.stringify(data)}`)
+    const { expires, originalMaxAge, ...options } = data
+    const cookie = new Cookie(options)
+    cookie.expires = expires ? new Date(expires) : undefined
+    cookie.originalMaxAge = originalMaxAge
+    return cookie
   }
-};
-
-/*!
- * Prototype.
- */
-
-Cookie.prototype = {
 
   /**
    * Set expires `date`.
    *
-   * @param {Date} date
-   * @api public
+   * @param {Date | null | undefined} date
+   * @public
    */
 
   set expires(date) {
-    this._expires = date;
-    this.originalMaxAge = this.maxAge;
-  },
+    this._expires = date || undefined
+    this.originalMaxAge = this.maxAge
+  }
 
   /**
-   * Get expires `date`.
+   * Get expires `Date` object to be the value for the `Expires Set-Cookie` attribute.
+   * By default, no expiration is set, and most clients will consider this a "non-persistent cookie" and will delete it on a condition like exiting a web browser application.
    *
-   * @return {Date}
-   * @api public
+   * @return {Date | undefined}
+   * @public
    */
 
   get expires() {
-    return this._expires;
-  },
+    return this._expires
+  }
 
   /**
    * Set expires via max-age in `ms`.
    *
-   * @param {Number} ms
-   * @api public
+   * @param {number | undefined} ms
+   * @public
    */
 
   set maxAge(ms) {
-    if (ms && typeof ms !== 'number' && !(ms instanceof Date)) {
-      throw new TypeError('maxAge must be a number or Date')
-    }
-
-    if (ms instanceof Date) {
-      deprecate('maxAge as Date; pass number of milliseconds instead')
+    if (ms !== undefined) {
+      if (typeof ms !== 'number') {
+        throw new TypeError('maxAge must be a number')
+      }
+      this.expires = new Date(Date.now() + ms)
+    } else {
+      this.expires = undefined
     }
-
-    this.expires = typeof ms === 'number'
-      ? new Date(Date.now() + ms)
-      : ms;
-  },
+  }
 
   /**
    * Get expires max-age in `ms`.
    *
-   * @return {Number}
-   * @api public
+   * @return {number | undefined}
+   * @public
    */
 
   get maxAge() {
     return this.expires instanceof Date
       ? this.expires.valueOf() - Date.now()
-      : this.expires;
-  },
+      : this.expires
+  }
 
   /**
    * Return cookie data object.
    *
-   * @return {Object}
-   * @api private
+   * @return {CookieSerializeOptions}
```

Comment:
This interface was and is still used for multiple purposes. The code isn't very clear who it's serving.

Specifying this type, it makes it clear it has a single purpose. As a result, the property could be renamed.

For now, I used this for `toJSON`, but it might be better to separate these two.

Reply:
Second iteration went back to the current API, which is a mix. As a result, it now uses `object` as type.